README.md

```diff
@@ -1,16 +1,171 @@
+our first milestone in learning to work with nested data.
 
-## SWBAT 2
+We're going to pick out three specific types of processing here:
+
+* Displaying the nested structure
+* Transforming the nested structure into a new structure
+* Transforming the nested structure into a result
+
+In each of these sections, we're going to give you a bit of "reference code."
+Reference code is code that's a little generic and probably doesn't solve any
+_real_ problem, but which is kept intentionally ***very simple*** so that you
+can see how it might be adapted to your particular need. Some people call these
+"reference implementations."
+
+## Display the cells in an `Array` of `Array`s
+
+It's not often we say this, but this following bit of code is worth memorizing.
+If you want to learn to speak any human language, it's a good idea to learn the
+grammar, grow your vocabulary and speak with native speakers as much as
+possible.  **But** that takes years, and, in a pinch, you should _really know_
+how to ask for food, water, and a bathroom. Even if you don't understand what's
+happening grammatically, having a few "stock phrases" to lean on is helpful.
+The following code is a good "stock phrase."
+
+Assuming the following AoA:
+
+```ruby
+spice_rack = [
+  ["Posh", "Scary", "Sporty"],
+  ["Paprika", "Fajita Mix", "Coriander"],
+  ["Parsley", "Sage", "Rosemary"]
+]
+```
+
+We display it like:
+
+```ruby
+spice_rack = [
+  ["Posh", "Scary", "Sporty"],
+  ["Paprika", "Fajita Mix", "Coriander"],
+  ["Parsley", "Sage", "Rosemary"]
+]
+
+row_index = 0
+while row_index < spice_rack.count do
+  element_index = 0
+  while element_index < spice_rack[row_index].count do
+    puts spice_rack[row_index][element_index]
+    element_index += 1
+  end
+  row_index += 1
+end #=> nil
+```
+
+This prints out:
+
+```text
+Posh
+Scary
+Sporty
+Paprika
+Fajita Mix
+Coriander
+Parsley
+Sage
+Rosemary
+```
+
+This "formula" can be expanded and changed. Maybe you want something more
+descriptive than `row_index` or `element_index` index. Or, perhaps you need to
+adjust behavior slightly. That's fine! But if you're not sure how to start,
+this is always a good start and will get you un-stuck!
+
+## Traverse `Array` of `Array`s to Produce a New Nested Data Structure
+
+The same "formula" that we suggested you use above can easily be adapted for
+doing work other than `puts`-ing things. Let's say we wanted to gather all the
+elements that start with `P`.
+
+```ruby
+spice_rack = [
+  ["Posh", "Scary", "Sporty"],
+  ["Paprika", "Fajita Mix", "Coriander"],
+  ["Parsley", "Sage", "Rosemary"]
+]
+
+
+results = []
+row_index = 0
+while row_index < spice_rack.count do
+  element_index = 0
+  while element_index < spice_rack[row_index].count do
+    if spice_rack[row_index][element_index][0] == "P"
+      results << spice_rack[row_index][element_index]
+    end
+    element_index += 1
+  end
+  row_index += 1
+end
+
+results #=> => ["Posh", "Paprika", "Parsley"]
+```
```

Comment:
This could use an explicit call out: `spice_rack[row_index][element_index][0] == "P"` Since the first two brackets are the coordinates, but the `[0]` is actually the index of the string character.